include/deal.II/base/mpi.h

```diff
@@ -1101,6 +1101,148 @@ namespace Utilities
            const T &          object_to_send,
            const unsigned int root_process = 0);
 
+
+    template <typename T>
+    struct is_mpi_type
+    {
+      static const bool value = false;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<bool>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<char>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<signed char>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<short>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<int>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<long int>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<unsigned char>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<unsigned short>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<unsigned int>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<unsigned long int>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<unsigned long long int>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<float>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<double>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<long double>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<std::complex<float>>
+    {
+      static const bool value = true;
+    };
+
+
+
+    template <>
+    struct is_mpi_type<std::complex<double>>
+    {
+      static const bool value = true;
+    };
```

Comment:
In https://github.com/dealii/dealii/issues/11955#issuecomment-969862409, @bangerth found a good idea to simplify this significantly using some more type traits.

Reply:
Please check the changes introduced in 81b1aa6